### DISCUSSION TITLE
Conditionally render different row

### DISCUSSION THREAD

**Discussion by mxthevs** [TanStack/table#4386] [Q&A]
I am trying to reproduce the following behavior, without much success 😅 :
![image](https://user-images.githubusercontent.com/46231311/190475178-8361a990-b3b7-49db-ac38-27b08a2796cc.png)

Is there a way to render a row like this based on some condition that I have in my data?

**Comment by writes-of-spring** [comment]
hmm, table's getCanExpand and using the sub component example should get you pretty far, if you omit the mechanism to expand which seems to be required based on your image, setting them all to be initially expanded may need thought, you could call toggleAllRowsExpanded(true) on mount, hopefully someone else has a more graceful suggestion!

https://tanstack.com/table/v8/docs/examples/react/sub-components
https://tanstack.com/table/v8/docs/api/features/expanding#getcanexpand
